ipapython/nsslib.py

```diff
@@ -129,6 +131,45 @@ def client_auth_data_callback(ca_names, chosen_nickname, password, certdb):
     socket.AF_UNSPEC: io.PR_AF_UNSPEC
 }
 
+
+def get_proper_tls_version_span(tls_version_min, tls_version_max):
+    # every tls version from `tls_versions` prior to min_allowed_idx
+    # is deprecated in IPA
+    min_allowed_idx = TLS_VERSIONS.index(TLS_VERSION_MINIMAL)
+
+    try:
+        min_version_idx = TLS_VERSIONS.index(tls_version_min)
+    except ValueError:
+        raise InvocationError("tls_version_min ('{val}') is not a known "
+                              "TLS version.".format(val=tls_version_min))
+
+    try:
+        max_version_idx = TLS_VERSIONS.index(tls_version_max)
+    except ValueError:
+        raise InvocationError("tls_version_max ('{val}') is not a known "
+                              "TLS version.".format(val=tls_version_max))
+
+    if min_version_idx > max_version_idx:
+        raise InvocationError("tls_version_min is higher than "
+                              "tls_version_max.")
+
+    if min_version_idx < min_allowed_idx:
+        min_version_idx = min_allowed_idx
+        root_logger.warning("tls_version_min set too low ('{old}'),"
```

Comment:
Please use a module-specific logger rather than the root logger:
```python
from ipapython.ipa_log_manager import log_mgr

logger = log_mgr.get_logger(__name__)
```